cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.certs.CertAndKey;
+import io.strimzi.certs.CertManager;
+import io.strimzi.certs.Subject;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class ClusterCa extends Ca {
+
+    // the Kubernetes service DNS domain is customizable on cluster creation but it's "cluster.local" by default
+    // there is no clean way to get it from a running application so we are passing it through an env var
+    public static final String KUBERNETES_SERVICE_DNS_DOMAIN =
+            System.getenv().getOrDefault("KUBERNETES_SERVICE_DNS_DOMAIN", "cluster.local");
+    private final String clusterName;
+    private Secret entityOperatorSecret;
+    private Secret topicOperatorSecret;
+
+    private Secret brokersSecret;
+    private Secret zkNodesSecret;
+
+    public ClusterCa(CertManager certManager, String clusterName, Secret caCertSecret, Secret caKeySecret) {
+        this(certManager, clusterName, caCertSecret, caKeySecret, 365, 30, true);
+    }
+
+    public ClusterCa(CertManager certManager,
+                     String clusterName,
+                     Secret clusterCaCert,
+                     Secret clusterCaKey,
+                     int validityDays,
+                     int renewalDays,
+                     boolean generateCa) {
+        super(certManager, AbstractModel.getClusterCaName(clusterName), clusterCaCert,
+                AbstractModel.getClusterCaKeyName(clusterName), clusterCaKey,
+                validityDays, renewalDays, generateCa);
+        this.clusterName = clusterName;
+    }
+
+    public void sort(List<Secret> secrets) {
+        Secret brokersSecret = null;
+        Secret eoSecrets = null;
+        Secret toSecret = null;
+        Secret zkNodesSecret = null;
+        for (Secret secret: secrets) {
+            String name = secret.getMetadata().getName();
+            if (KafkaCluster.brokersSecretName(clusterName).equals(name)) {
+                brokersSecret = secret;
+            } else if (EntityOperator.secretName(clusterName).equals(name)) {
+                eoSecrets = secret;
+            } else if (TopicOperator.secretName(clusterName).equals(name)) {
+                toSecret = secret;
+            } else if (ZookeeperCluster.nodesSecretName(clusterName).equals(name)) {
+                zkNodesSecret = secret;
+            }
+        }
+        this.brokersSecret = brokersSecret;
+        this.entityOperatorSecret = eoSecrets;
+        this.topicOperatorSecret = toSecret;
+        this.zkNodesSecret = zkNodesSecret;
+    }
+
+    public Secret topicOperatorSecret() {
```

Comment:
Should this be `getTopicOperatorSecret`? Same applies for the next method. Juts a question, feel free to ignore.

Reply:
Kafka itself tends to avoid using the getter/setter pattern. I don't think many/any of its classes are intended to be used with frameworks which rely on that patter, so that's OK. So in strimzi we can chose whether to follow the pattern religiously or not, and no matter which we chose we're at odds with either the Java convention or the Kafka one. Or we can decide to use getter/setters when that's needed by a framework we use (like Jackson in the `api`). Doing it when needed has some merit: it's clearer when there's some framework which will be introspecting on the code.

At this point I don't really care either way, but whatever the community decides, we should be consistent.

Reply:
tbh I really love to avoid getter/setter and using this pattern.